File: Programmers/java/lev2/구명보트/solution.java
import java.util.Deque;
import java.util.ArrayDeque;
import java.util.Arrays;

class Solution {
    public int solution(int[] people, int limit) {
        int cnt = 0 , weight = 0;
        Arrays.sort(people);
        
        Deque<Integer> deque = new ArrayDeque<>();
        for (int i : people )
            deque.add(i);
        
        while( deque.size() > 1){
            weight = deque.getFirst()+ deque.getLast();

            if (weight <= limit){
                deque.removeFirst();
                deque.removeLast();
            }
            else if (weight > limit)
                deque.removeLast();

            cnt++;

            if ( deque.size()==1 )
                cnt++;
        }
        return cnt;
    }
}

File: Programmers/py/lev2/구명보트/solution.py
from collections import deque
def solution(people, limit):
    people = deque( sorted(people))
    cnt = 0
   
    while( len(people) > 1):
        weight = people[0] + people[-1]
        
        if weight <= limit:
            people.popleft()
            people.pop()
            
        elif weight > limit:
            people.pop()
        
        cnt+=1
        
        if len(people)==1:
            cnt+=1
            
    return cnt
        
